scout/adapter/mongo/variant.py

```diff
@@ -469,69 +471,94 @@ def _find_affected(self, case_obj):
             limit_genes (list): list of gene hgnc_ids to limit the search to
 
         Returns:
-            causatives(iterable(Variant))
+            (iterable(Variant))
         """
 
-        if len(positional_variant_ids) == 0:
-            return []
-
         filters = {"variant_id": {"$in": list(positional_variant_ids)}}
-        if case_obj:
-            affected_ids = self._find_affected(case_obj)
-            if len(affected_ids) == 0:
-                return []
-
-            filters["case_id"] = case_obj["_id"]
-            filters["samples"] = {
-                "$elemMatch": {
-                    "sample_id": {"$in": affected_ids},
-                    "genotype_call": {"$regex": "1"},
-                }
+
+        affected_ids = self._find_affected(case_obj)
+        if len(affected_ids) == 0:
+            return []
+        filters["case_id"] = case_obj["_id"]
+        filters["samples"] = {
+            "$elemMatch": {
+                "sample_id": {"$in": affected_ids},
+                "genotype_call": {"$regex": CARRIER},
             }
-        else:
-            filters["institute"] = institute_obj["_id"]
+        }
+
         if limit_genes:
             filters["genes.hgnc_id"] = {"$in": limit_genes}
 
         return self.variant_collection.find(filters)
 
-    def check_causatives(self, case_obj=None, institute_obj=None, limit_genes=None):
-        """Check if there are any variants that are previously marked causative
+    def institute_causatives(self, institute_obj, limit_genes=None):
+        """Return all causative variants for an institute - eventually within the provided list of genes
+
+        Args:
+            institute_obj (dict): an Institute object
+            limit_genes (list): list of gene hgnc_ids to limit the search to
+
+        Returns:
+            iterable(Variant)
+        """
+        causative_events = self.event_collection.find(
+            {
+                "institute": institute_obj["_id"],
+                "verb": {"$in": ["mark_causative", "mark_partial_causative"]},
+                "category": "case",
+            }
+        )
+        causative_ids = set()
+        for case_event in causative_events:
+
+            case_obj = self.case(case_event.get("case"))
+            if case_obj is None:
+                continue
+
+            for var_id in case_obj.get("causatives", []):
+                causative_ids.add(var_id)
 
-        Loop through all variants that are marked 'causative' for an
-        institute and check if any of the variants are present in the
-        current case.
+            for var_id, _ in case_obj.get("patial_causatives", {}).items():
+                causative_ids.add(var_id)
+
+        filters = {"_id": {"$in": list(causative_ids)}}
+        if limit_genes:
+            filters["genes.hgnc_id"] = {"$in": limit_genes}
+
+        return self.variant_collection.find(filters)
+
+    def case_matching_causatives(self, case_obj=None, limit_genes=None):
```

Comment:
This is the old `check_causatives`, that now is used only to retrieve matching causatives from other cases

Reply:
You should probably ditch the `None` defaults here? It would just crash with `case_obj == None` wouldn't it?

Reply:
Right! Thanks!